tests/virt_autotest/virsh_external_snapshot.pm

```diff
@@ -0,0 +1,59 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Test VM external snapshot using virsh (create - restore - delete)
+# Maintainer: Leon Guo <xguo@suse.com>
+#
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+
+sub run {
+    #Snapshots are supported on KVM VM Host Servers only
+    return unless check_var("REGRESSION", "qemu-hypervisor") || check_var("SYSTEM_ROLE", "kvm");
+
+    my $wait_script                 = "30";
+    my $get_vm_hostnames_inactive   = "virsh list --inactive | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames_inactive       = script_output($get_vm_hostnames_inactive, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
+
+    foreach (@vm_hostnames_inactive_array)
+    {
+        assert_script_run("virsh start $_");
+    }
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "virsh-snapshot", "Creating External Snapshot of guest's disk";
+        assert_script_run "virsh snapshot-create-as $guest --disk-only --diskspec vda,snapshot=external,file=/var/lib/libvirt/images/$guest.disk-only --name external-snapshot-$guest";
+        assert_script_run "virsh snapshot-list $guest | grep external-snapshot-$guest";
+
+        record_info "virsh-snapshot", "Creating Live External Snapshot of guest's memory and disk state";
+        assert_script_run "virsh snapshot-create-as $guest --live --memspec snapshot=external,file=/var/lib/libvirt/images/$guest.memspec --diskspec vda,snapshot=external,file=/var/lib/libvirt/images/$guest.diskspec --name external-snapshot-live-$guest";
+        assert_script_run "virsh snapshot-list $guest | grep external-snapshot-live-$guest";
+
+        record_info "virsh-snapshot", "SKIP - No support to start external snapshot";
+
+        record_info "virsh-snapshot", "SKIP - No support to delete external snapshot";
```

Comment:
Why are here those `record_info`'s without content they record?

Reply:
Because of so far libvirt totally had not support to start and delete  external snapshot, so here just only informational purposes without any  content they record via record_info.